src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5455,6 +5467,12 @@ private void ParseTypeArgumentList(out SyntaxToken open, SeparatedSyntaxListBuil
             }
 
             close = this.EatToken(SyntaxKind.GreaterThanToken);
+
```

Comment:
please place a 'return' here to help separate out the normal code flow, and hte helper extension methods.